examples/react-signer/src/service/method/interactive/icrc34-delegation-method.service.ts

```diff
@@ -90,25 +66,51 @@ class Icrc34DelegationMethodService extends InteractiveMethodService {
       isPublicAccountsAllowed,
     }
   }
+  private formatDelegationChain(chain: DelegationChain) {
+    return {
+      signerDelegation: chain.delegations.map((signedDelegation) => {
+        const { delegation, signature } = signedDelegation
+        const { targets } = delegation
+        return {
+          delegation: Object.assign(
+            {
+              expiration: delegation.expiration,
+              pubkey: toBase64(delegation.pubkey),
+            },
+            targets && {
+              targets: targets.map((t) => t.toText()),
+            }
+          ),
+          signature: toBase64(signature),
+        }
+      }),
+      publicKey: toBase64(chain.publicKey),
+    }
+  }
+
+  private async isPublicAccountsAllowed(targets: string[], origin: string) {
+    console.log("isPublicAccountsAllowed")
+    if (!targets || targets.length === 0) return false
+
+    console.log("isPublicAccountsAllowed")
+    try {
+      await targetService.validateTargets(targets, origin)
+      console.log("isPublicAccountsAllowed true")
+      return true
+    } catch (e: unknown) {
+      const text = e instanceof Error ? e.message : "Unknown error"
+      console.error("The targets cannot be validated:", text)
```

Comment:
just in case, shouldn't we show this validation message to user?

Reply:
if so, just pass error to radio component, i will add styles later

Reply:
That's another improvement, we don't show the message to the user but do not allow him to pick a public account. And send an error to logs.